src/utils/lazyRetry.ts

```diff
@@ -0,0 +1,37 @@
+import type {ComponentType} from 'react';
+
+type ComponentImport<T> = () => Promise<{default: ComponentType<T>}>;
+
+/**
+ * Attempts to lazily import a React component with a retry mechanism on failure.
+ * If the initial import fails with a `ChunkLoadError`, the function will refresh the page once and retry the import.
+ * If the import fails again after the refresh, the error is propagated.
+ *
+ * @param componentImport - A function that returns a promise resolving to a lazily imported React component.
+ * @returns A promise that resolves to the imported component or rejects with an error after a retry attempt.
+ */
+const lazyRetry = function <T>(componentImport: ComponentImport<T>): Promise<{default: ComponentType<T>}> {
+    return new Promise((resolve, reject) => {
+        // Retrieve the retry status from sessionStorage, defaulting to 'false' if not set
+        const hasRefreshed: unknown = JSON.parse(sessionStorage.getItem('retry-lazy-refreshed') ?? 'false');
+
+        componentImport()
+            .then((component) => {
+                // Reset the retry status to 'false' on successful import
+                sessionStorage.setItem('retry-lazy-refreshed', 'false'); // success so reset the refresh
+                resolve(component);
+            })
+            .catch((error) => {
+                if (!hasRefreshed) {
```

Comment:
```suggestion
                if (!hasRefreshed) {
                console.error('Failed to lazily import a React component, refreshing the page in order to retry the operation.', component);
```

What about adding a console error here?

Reply:
Yeah, great idea

Added